server/src/main/java/org/opensearch/search/aggregations/metrics/SumIntegralAggregator.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+/*
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.search.aggregations.metrics;
+
+import org.apache.lucene.index.LeafReaderContext;
+import org.apache.lucene.index.SortedNumericDocValues;
+import org.apache.lucene.search.ScoreMode;
+import org.opensearch.common.lease.Releasables;
+import org.opensearch.common.util.BigArrays;
+import org.opensearch.common.util.DoubleArray;
+import org.opensearch.search.DocValueFormat;
+import org.opensearch.search.aggregations.Aggregator;
+import org.opensearch.search.aggregations.InternalAggregation;
+import org.opensearch.search.aggregations.LeafBucketCollector;
+import org.opensearch.search.aggregations.LeafBucketCollectorBase;
+import org.opensearch.search.aggregations.support.ValuesSource;
+import org.opensearch.search.aggregations.support.ValuesSourceConfig;
+import org.opensearch.search.internal.SearchContext;
+
+import java.io.IOException;
+import java.math.BigInteger;
+import java.util.Map;
+
+/**
+ * Aggregate all docs into a single sum value for integral numbers
+ *
+ * @opensearch.internal
+ */
+public class SumIntegralAggregator extends NumericMetricsAggregator.SingleValue {
+
+    private final ValuesSource.Numeric valuesSource;
+    private final DocValueFormat format;
+    private DoubleArray sums;
+
+    SumIntegralAggregator(
+        String name,
+        ValuesSourceConfig valuesSourceConfig,
+        SearchContext context,
+        Aggregator parent,
+        Map<String, Object> metadata
+    ) throws IOException {
+        super(name, context, parent, metadata);
+        this.valuesSource = (ValuesSource.Numeric) valuesSourceConfig.getValuesSource();
+        this.format = valuesSourceConfig.format();
+        sums = context.bigArrays().newDoubleArray(1, true);
```

Comment:
If this is an integral list, wouldn't we be able to get a newLongArray from this?  I'm concerned that we need to convert each bucket into a double value.

Reply:
Because of the way that summation is integrated, it has to return a double, meaning that if the final long value is too long it will lose precision but any arithmetic done in the summation with large numbers will be done correctly as long as the final value is small enough to be accurately held by the double. I can't change the output to a long array because if the final summation is larger than the value that can be held by a long then it wouldn't work.

Reply:
Good point, fixed in https://github.com/opensearch-project/OpenSearch/pull/6984/commits/9eead608735c284f0b3bce5301400f8bcd24de1d